src/popup/pages/NetworkForm.vue

```diff
@@ -149,6 +150,21 @@ export default defineComponent({
       };
     }
 
+    async function fetchNetworkId() {
+      try {
+        const node = await Node({ url: newNetwork.value.url });
+        if (!(newNetwork.value as any).networkId) {
+          root.$store.commit('setUserNetwork', {
+            ...newNetwork.value,
+            index: networks.value[newNetwork.value.name].index,
+            networkId: node.nodeNetworkId,
+          });
+        }
+      } catch (e) {
+        handleUnknownError(e);
+      }
```

Comment:
What is the motivation for defining `Node` once again? You could use the SDK API.

Reply:
I also don't like this piece of code. For me updating the user networks data with data fetched from the node is rather bad thing.